thunder/torch/__init__.py

```diff
@@ -1286,7 +1287,7 @@ def transpose(a: TensorLike, /, dim0: int, dim1: int) -> TensorLike:
 @torchsymbol(torch.unbind, is_method=True)
 def unbind(a: TensorLike, /, dim: int = 0) -> tuple[TensorLike, ...]:
     utils.check(
-        len(a.size()) > 0,
+        a.ndim > 0,
```

Comment:
I want to point it out.

There's a subtle difference between the old `len(a.size())` vs `a.ndim`. calling `a.size()` result in a `prims.shape` in the old trace, even though it's being DCE'ed away, that logic didn't kick in before fusion claims each operation.

Since nvfuser doesn't take `prims.shape` yet, it would reject the node, causing `pytest tests/test_nvfuser.py -v -s -k test_cse_rematerialization_nvfuser_cuda_None` to fail because we have more fusion symbols in the test.

Reply:
I have no objections to the solution, but from my pov it would be fine to dce at the top of transform for execution if that helps.
(ps thanks for keeping the shape prim's name shape. :slightly_smiling_face: )